spec/actions_manager_spec.coffee

```diff
@@ -286,34 +284,64 @@ describe 'ActionsManager', ->
 
         action_reporting_tests()
 
-        context 'when action is reported successfully', ->
+        context 'when redirect callback is supplied', ->
           beforeEach ->
-            @run()
-            @sendbeacon_promise.resolve()
-            return
+            @redirect_url = 'http://redirect_url'
+            @redirect_callback_spy = sinon.spy()
 
-          it 'redirects ', ->
-            expect(@settings.redirectTo).to.be.called
+            @run = ->
+              sa(@category, @type, @data, @redirect_callback_spy, @redirect_url)
+              @initializeSubject()
+              @subject.run()
 
-          it 'redirects to redirect_url', ->
-            expect(@settings.redirectTo.args[0][0]).to.contain @redirect_url
+          context 'when action is reported successfully', ->
+            it 'executes redirect callback', ->
+              @run()
+              @sendbeacon_promise.resolve()
 
-        context 'when action fails to report', ->
-          beforeEach ->
-            @run()
-            @sendbeacon_promise.reject()
-            return
+              expect(@redirect_callback_spy).to.be.calledWith(@redirect_url)
+
+            context 'when redirect callback throws an error', ->
+              beforeEach ->
+                @redirect_callback_stub = sinon.stub().throws()
+
+                @run = ->
+                  sa(@category, @type, @data, @redirect_callback_stub, @redirect_url)
+                  @initializeSubject()
+                  @subject.run()
+
+                @run()
+                @sendbeacon_promise.resolve()
+                return
+
+              it 'redirects', ->
```

Comment:
I know this was here before, but do we really need two tests for this?

Reply:
I know this was here, ~before it was cool~ 🤣 

You are right, i will remove it.

Reply:
Done!